src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php

```diff
@@ -322,15 +322,21 @@ public static function getClassDescription(string $class, string &$resolvedClass
 
     private function getContainerEnvVars(ContainerBuilder $container): array
     {
+        $file = file_get_contents($container->getParameter('debug.container.dump'));
+
+        preg_match_all('{%env\((.*)\)%}', $file, $envVars);
+        $envVars = array_unique($envVars[1]);
+
         $getEnvReflection = new \ReflectionMethod($container, 'getEnv');
         $getEnvReflection->setAccessible(true);
-        $envs = [];
-        foreach (array_keys($container->getEnvCounters()) as $env) {
+
+        foreach ($envVars as $env) {
             $processor = 'string';
             if (false !== $i = strrpos($name = $env, ':')) {
                 $name = substr($env, $i + 1);
                 $processor = substr($env, 0, $i);
             }
+            // dump($container->getParameter('env(DATABASE_URL)'));die;
```

Comment:
you should capture e.g. `<parameter key="env(DATABASE_URL)"></parameter>` as well